runtime-common/realm.ts

```diff
@@ -144,17 +134,39 @@ export class Realm {
 
   async #startup() {
     await Promise.resolve();
-    await this.#searchIndex.run();
+    this.#searchIndex.run();
+    // The first phase will perform a from scratch index, but leave the docs
+    // with links in an error state since in order to prevent a deadlock since
+    // the index can't query it's own realm's index (aka itself) via fetch
+    // before it starts up.
+    await this.#searchIndex.phase1Ready();
+
+    // After the 1st indexing phase, the realm can answer index based requests
+    // from itself, and notably return with errors docs when asked about an
+    // instance with an unloaded link.
+    this.#readyForLocalRequests = true;
+
+    // The 2nd phase of indexing invalidates all the instances that have
+    // unloaded links and is now able to load the unloaded links
+    await this.#searchIndex.ready();
```

Comment:
In order to prevent deadlocking during search index building from scratch, I had to separate the initial search indexing into 2 phases

Reply:
this feels very similar to our generation solution in the older implementations--we should use that approach